evm/chain.py

```diff
@@ -130,8 +130,8 @@ def get_vm_class_for_block_number(self, block_number):
         for n in reversed(self.vms_by_range.keys()):
             if block_number >= n:
                 return self.vms_by_range[n]
-        raise EVMNotFound(
-            "There is no EVM available for block #{0}".format(block_number))
+        raise ChainNotFound(
+            "There is no Chain available for block #{0}".format(block_number))
```

Comment:
Just noting that: s/Chain/VM

This can be fixed in a separate small PR as I'd like to get this merged ASAP because it's likely to cause merge conflicts for other PRs.